### ISSUE TITLE
Core/Battle.net: Implement time-based token for user login

### ISSUE THREAD

**Issue by Luzifix** [AshamaneProject/AshamaneCore#995]
**Description:**
Currently it is not possible to use an authenticator in the bnetserver to increase the security of the accounts

**Current behaviour:**
The bnetserver currently not able able to ask to the user to enter a digital code from Authentication before login

**Expected behaviour:**
The bnetserver should be able to ask to the user to enter a code from Authenticator beforelog in

**Project(s): <-- Mandatory**
- [x] Issue found playing in Live Ashamane Private Server.
- [x] Issue found in a compiled master repo from Git.

**TC rev. hash/commit:** 
29755f86c0212fbe4ed739ed7889ea32dd0c05c6

**Database version**
ADB735.00

**Operating system:** Windows, Linux

**Comment by LuigiElleBalotta** [comment]
cool!

**Comment by Luzifix** [comment]
Yes i think a lot of Server wish to support more battle.net features like Authenticator, Friendsystem and Crossrealm
